gravityflow/gravityflow
# Fix whitespace text getting accepted for a 'Required' Workflow Note

## Description
Closes [Issue# 92](https://github.com/gravityflow/backlog/issues/92)
If a Workflow Note is marked as 'Required' in Workflow Step Settings, it should not accept whitespace as valid input.

## Testing instructions
1. Create a Workflow with Approval and User Input Steps.
2. On Workflow Step Settings enable Workflow Note as 'required'.
3. Check to complete the workflow step in Workflow Inbox with whitespace as the 'Note' input.
4. Now switch to this branch, confirm submitting a note with whitespace does yield the validation, 'A note is required'.

## Automated Test Enhancements
Added test `0055WorkflowNoteValidationCept.php` to confirm workflow note validation for a whitespace character note.
 
## Screenshots
N/A

## Documentation Changes
Added filter `gravityflow_note_valid` to customize note validity
```
add_filter( 'gravityflow_note_valid', 'my_gravityflow_note_valid', 10, 4 );
function my_gravityflow_note_valid( $valid, $note, $new_status, $step ) {
	
	// for a particular step, only a specific note can make the step note acceptable
	if ( $step->get_id() == 74 ) {
		if ( $note == 'Accepted' ) {
			$valid = true;
		}
		else {
			$valid = false;
		}
	}

	return $valid;
}
```

## Checklist:
